src/cocotb/types/_logic.py

```diff
@@ -97,7 +96,7 @@ class Logic:
 
     .. code-block:: pycon3
 
-        >>> def full_adder(a: Logic, b: Logic, carry: Logic) -> Tuple[Logic, Logic]:
+        >>> def full_adder(a: Logic, b: Logic, carry: Logic) -> tuple[Logic, Logic]:
```

Comment:
This was done because of how `--doctest-modules` tests doctests. It evaluates the module where the object is defined at before running the test to pull in any dependencies for testing. Since we are fixing up the `__module__` this is breaking which module gets evaluated so it's `cocotb/types/__init__.py` instead of `_logic.py`. 

Luckily since we are building docs with only Python > 3.11 we don't even need this import anymore.

Reply:
Okay so Python 3.8 doesn't like that lmao.